src/Value/CSSString.php

```diff
@@ -53,7 +53,7 @@ public static function parse(ParserState $oParserState): CSSString
         $sContent = null;
         if ($sQuote === null) {
             // Unquoted strings end in whitespace or with braces, brackets, parentheses
-            while (!\preg_match('/[\\s{}()<>\\[\\]]/isu', $oParserState->peek())) {
+            while (\in_array(\preg_match('/[\\s{}()<>\\[\\]]/isu', $oParserState->peek()), [0, false], true)) {
```

Comment:
This could be `\preg_match(...) !== 1`, though I'd prefer `\preg_match(...) < 1` if possible, in case it's later changed to `preg_match_all()`.

The Rector-proposed change makes the code unnecessarily unreadable, and perhaps should be reported as a bug there.

The mixed return types for the built in PHP functions do cause some headaches with static analysis tools, which was why I ended up implementing https://github.com/MyIntervals/emogrifier/pull/1319 and others, having originally only intended https://github.com/MyIntervals/emogrifier/pull/1308 to provide a better fallback for the `replace` functions.  Maybe that class can be moved to a separate library...

Reply:
Done. (I've used `!== 1` in order to avoid potentially using `<` on `false`.)